elastic/kibana
# [Fleet] Add permissions checks

## Summary

Adds permissions checks to the Fleet UI, resolves https://github.com/elastic/kibana/issues/48032

### To be done in future PR

- [ ] Add icon to permissions registration (needs icon to be created)

### To test

1) As a super user create 3 user roles in Kibana
  a)  A role with fleet read permission on all spaces
  b)  A role with fleet write permissions on all spaces
  c) A role with no 
2) Create 3 new users each with one of the above created roles
3) Login with user containing role `A` -- Expect to see fleets UI without a button to enroll agents
4) Login with user containing role `B` -- Expect to see fleets UI **WITH** a button to enroll agents
5) Login with user containing role `C` -- Expect to see an access denied error when attempting to access fleet.
